tracer/src/Datadog.Trace/Debugger/PInvoke/DebuggerNativeMethods.cs

```diff
@@ -0,0 +1,91 @@
+// <copyright file="DebuggerNativeMethods.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Datadog.Trace.Debugger.ProbeStatuses;
+
+namespace Datadog.Trace.Debugger.PInvoke
+{
+    internal static class DebuggerNativeMethods
+    {
+        public static void InstrumentProbes(NativeMethodProbeDefinition[] methodProbes, NativeLineProbeDefinition[] lineProbes, NativeRemoveProbeRequest[] revertProbes)
```

Comment:
May be we should check, if `methodProbes` is not empty, before calling native code?
Same for other values?

Reply:
It doesn't matter if `methodProbes` is empty/not as you also have line probes and revert requests. Although hoping into native is costly it happens per probe configuration change which is not that frequent.
Let's have that discussion internally.

Reply:
Oh, I see.
But there is still small chance, that all method probes, line probes and revert probes are all zero, in case that all probes we got are `unbound`.
I'm not sure what is the chance, when all works fine, but for, in case of some weird bug, it could happen.
I think we can add simple `if` check for this case. Just in case.

Reply:
Not a stopper, of course, just a suggestion.

Reply:
Let's revisit it later, I don't think we should address it now considering the low chances of it to ever happen and the necessity of pushing this PR into master. Not only it's not a stopper, we have many other fixes that are way higher in the list before we ever reach this one.